### PR TITLE
Private Clocks

### PR DESCRIPTION
**Pull request by buttonboy**
Added ability for GMs to create private clocks that can not be viewed by players.

- There is a checkbox on the edit dialog to make a clock private.
- An eye slash icon will appear next to the name of a clock to inform the GM that it is private.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Thanks for doing this. This seems against the spirit of clocks, but it can't hurt.

Can you change how visibility is set using a hidden css class? Something like `{{#if (and clock.private @root.editable)}}hidden{{/if}}`. If that doesn't mess with sortable then it would be much preferred as a solution.

**Comment by buttonboy** [comment]
Not a problem. That's actually the first way I did it. I was just slightly concerned a skilled player would be able to unhide the private clock. However the css class approach is much simpler code. I was unfortunatly unable to figure out how to do an 'and' with handlebarjs and could not find any documentation.  So I settled on an 'unless' nessed in an 'if'. I hope this will suffice.

Sorting works fine with this approach. The only noticable diffrence is there is not animation when hiding or unhiding a clock. But in reality I don't think that will happen very often.

My GM likes to use clocks for various things like tracking if an encounter is close to happening. So hopefully other GMs will find uses for private clocks.

File: templates/clock-panel.hbs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <section class="clock-panel {{#if @root.options.editable}}editable{{/if}} {{verticalEdge}}">
     <div class="clock-list">
-        {{#each clocks as |clock|}}
-            <div class="clock-entry" data-id="{{clock.id}}">
+         {{#each clocks as |clock|}}
```
Comment (CarlosFdez) [inline templates/clock-panel.hbs:3]:
```suggestion
        {{#each clocks as |clock|}}
```

File: templates/clock-panel.hbs [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+            <div class="clock-entry {{#if clock.private}}{{#unless @root.options.editable}}hidden{{/unless}}{{/if}}" data-id="{{clock.id}}">
```
Comment (CarlosFdez) [inline templates/clock-panel.hbs:4]:
```suggestion
            <div class="clock-entry {{#if (and clock.private (not @root.options.editable)}}hidden{{/if}}" data-id="{{clock.id}}">
```

**Comment by buttonboy** [comment]
Awesome! Thank you! It looks like that is working.

**Comment by CarlosFdez** [comment]
Thank you. I wont' release this immediately, but it'll be in there when I finally get to finishing the clock theming stuff.

**Pull request merged**